nx/lib/nx/binary_backend.ex

```diff
@@ -829,6 +829,26 @@ defmodule Nx.BinaryBackend do
   @impl true
   def conjugate(out, tensor), do: element_wise_unary_op(out, tensor, &Complex.conjugate/1)
 
+  @impl true
+  def real(out, tensor) do
+    real_fn = fn
+      %Complex{re: re} -> re
+      num -> num
+    end
+
+    element_wise_unary_op(out, tensor, real_fn)
```

Comment:
Per my comments on Nx, this should only be called if the tensor is a float and we can actually specialize the implementation: we traverse the binary getting only the real parts and write it down as float. We don't need to create the structs.